pkg/sync/http/http_sync.go

```diff
@@ -0,0 +1,140 @@
+package http
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha1" //nolint:gosec
+	"encoding/base64"
+	"errors"
+	"io"
+	"net/http"
+
+	"github.com/open-feature/flagd/pkg/sync"
+
+	"github.com/open-feature/flagd/pkg/logger"
+)
+
+type Sync struct {
+	URI          string
+	Client       Client
+	Cron         Cron
+	BearerToken  string
+	LastBodySHA  string
+	Logger       *logger.Logger
+	ProviderArgs sync.ProviderArgs
+}
+
+// Client defines the behaviour required of a http client
+type Client interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// Cron defines the behaviour required of a cron
+type Cron interface {
+	AddFunc(spec string, cmd func()) error
+	Start()
+	Stop()
+}
+
+func (hs *Sync) Sync(ctx context.Context, dataSync chan<- sync.DataSync) error {
+	// Initial fetch
+	fetch, err := hs.Fetch(ctx)
+	if err != nil {
+		hs.Logger.Error(err.Error())
```

Comment:
Is returning an error after handling it (with a log) the right thing to do?

Reply:
Right now, `Runtime` setup an error group and adds sync impl to routines. The initial fetch on the other hand makes sure our source is correct. And if fetch fails, then we return this error essentially to the error group so that it could exit [1]. I added log here for clarity, so that we do not need explicit error handing at `Ruintime` and rely on other logs from error group internals. 

I am open for improvements here

[1] - https://pkg.go.dev/golang.org/x/sync/errgroup#Group.Go

Reply:
Sure, I updated to not to log with the latest commit